### PR TITLE
Fix issues with verification case 1.e

### LINKED ISSUE #88: Issues with verification test 1.e
## Bug Description
1. Poor solve parameters are causing oscillations in the output. This was also referenced by @RemDelaporteMathurin in #73.
2. [Vector postprocessor](https://github.com/idaholab/TMAP8/blob/c711bc80a10bddee5319d0d560a10a63dd5346e3/test/tests/ver-1e/ver-1e.i#L50) is commented out, but it is required by the Python script for checking steady state solution. Currently the input file will not produce the expected csv file name for the steady state solution verification.
3. [PostProcessor](https://github.com/idaholab/TMAP8/blob/c711bc80a10bddee5319d0d560a10a63dd5346e3/test/tests/ver-1e/ver-1e.i#L65) required by Python script is commented out too.
4. [Outputs block](https://github.com/idaholab/TMAP8/blob/c711bc80a10bddee5319d0d560a10a63dd5346e3/test/tests/ver-1e/ver-1e.i#L94) needs csv = True, currently it only outputs an exodus file.
5. Python script uses an analytical solution CSV file instead of actually solving the analytical solution.

## Impact
This verification test requires the user to make significant modifications to the MOOSE input file and potentially the Python script in order to get any usable results. The verification tests should run as they are.

### PR DESCRIPTION
**Pull request by ChaitanyaBhave-e**
# Closes #88 and #73 

## Reason
Fixes several issues that made this verification case not run out of the box. Additionally added extra heavy test cases for exodus file and analytical solutions as per the new standards. However, more work will need to happen in a follow up PR to clean up the documentation further and provide additional information.

### CODE REVIEW

**Comment by ChaitanyaBhave-e** [comment]
@simopier @RemDelaporteMathurin

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 1]
```diff
@@ -4,28 +4,50 @@
...
+tmap_sol = pd.read_csv("./gold/ver-1e_csv.csv")
 tmap_time = tmap_sol['time']
 tmap_conc = tmap_sol['conc_point1']
 
-analytical_sol = pd.read_csv("./analytical_time.csv")
-analytical_time = analytical_sol['t']
-analytical_conc = analytical_sol['u']
+#Analytical solution
+t = np.expand_dims(tmap_time,axis=0)
+x = 15.75e-6
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:21]:
```suggestion
x = 15.75e-6           # distance into the SiC layer of a PyC/SiC composite structure (microns)   
```

Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:21]: This is for the TMAP7 case, which is different than the TMAP4 case (8 microns), could you include both here and in the tests, and update the documentation to discuss the difference?

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 2]
```diff
@@ -4,28 +4,50 @@
...
 
-tmap_sol = pd.read_csv("./ver-1e_csv.csv")
+tmap_sol = pd.read_csv("./gold/ver-1e_csv.csv")
 tmap_time = tmap_sol['time']
 tmap_conc = tmap_sol['conc_point1']
 
-analytical_sol = pd.read_csv("./analytical_time.csv")
-analytical_time = analytical_sol['t']
-analytical_conc = analytical_sol['u']
+#Analytical solution
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:19]:
```suggestion
# Analytical solution
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 3]
```diff
@@ -4,28 +4,50 @@
...
+x = 15.75e-6
+c0 = 50.7079            # concentration at the PyC free surface (moles/um^3)
+a  = 33e-6              # thickness of the PyC layer (um)
+l  = 66e-6              # thickness of the SiC layer (um)
+D_PyC = 1.274e-7        # diffusivity in PyC (m^2/s)
+D_SiC = 2.622e-11       # diffusivity in SiC (m^2/s)
+
+k = sqrt(D_PyC/D_SiC)
+
+#Calculate lambda values for analytical solution
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:30]:
```suggestion
# Calculate lambda values for analytical solution
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 4]
```diff
@@ -4,28 +4,50 @@
...
+lambda_range = np.arange(1e-12,1e5,1e-4)
+f = 1/tan(lambda_range*a)
+g = 1/tan(k*l*lambda_range)/k
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:33]: Yep, that's what we talked about: the equation in ver-1e in TMAP7 was incorrect, whereas TMAP4 was correct. Could you make a note in the documentation (like I did for ver-1a) to explain the typo in the TMAP7 V&V report to users?

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 5]
```diff
@@ -39,16 +61,22 @@
...
-analytical_distance = analytical_sol['x']*1e6
-analytical_conc = analytical_sol['u']
+x = tmap_distance
+PyC_conc = c0*(1 + (x/l)*((a*D_PyC)/(a*D_PyC + l*D_SiC) - 1 ) )
+SiC_conc = c0*(((a+l-x)/l)*(a*D_PyC)/(a*D_PyC + l*D_SiC) )
+analytical_conc = (x<a)*PyC_conc+(x>=a)*SiC_conc
+
+RMSE = np.sqrt(np.mean((tmap_conc-analytical_conc)**2))
+err_percent = RMSE*100/np.mean(analytical_conc)
+print(err_percent)
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:76]: Could you add this value as a printed `R=xxxx` in the figure so that it appears in the documentation?

File: test/tests/ver-1e/tests [hunk 6]
```diff
@@ -1,12 +1,48 @@
...
+  [ver-1e]
     type = Exodiff
     input = ver-1e.i
-    exodiff = ver-1e_out.e
-    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition.'
+    exodiff = ver-1e_test_exodus.e
+    cli_args = 'Executioner/dt=0.2 Executioner/end_time=50 Outputs/exodus/file_base=ver-1e_test_exodus'
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source.'
     exodiff_opts = '-match_ids'
     map = False
```
Comment (simopier) [inline test/tests/ver-1e/tests:11]: I honestly don't know what these do. Could you educate me?

File: test/tests/ver-1e/tests [hunk 7]
```diff
@@ -1,12 +1,48 @@
...
   design = 'Diffusion.md TimeDerivative.md DirichletBC.md'
   issues = '#12'
-  [csv]
+  [ver-1e]
     type = Exodiff
     input = ver-1e.i
-    exodiff = ver-1e_out.e
-    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition.'
+    exodiff = ver-1e_test_exodus.e
+    cli_args = 'Executioner/dt=0.2 Executioner/end_time=50 Outputs/exodus/file_base=ver-1e_test_exodus'
```
Comment (simopier) [inline test/tests/ver-1e/tests:8]: Could you add a statement in the documentation (like I did for ver-1a) to explain that the test is different from the verification run?

**Review by simopier — CHANGES_REQUESTED** [review]
Thank you for this fix!
I left a couple of comments throughout the code, and have an additional request:
- Could you update the documentation with the new figure that you produce with the updated python script?

File: test/tests/ver-1e/comparison_ver-1e.py (hunk 1 above)
Comment (RemDelaporteMathurin) [reply test/tests/ver-1e/comparison_ver-1e.py:21]: Actually the distance is in meters here not microns

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 8]
```diff
@@ -4,28 +4,50 @@
...
+lambda_range = np.arange(1e-12,1e5,1e-4)
+f = 1/tan(lambda_range*a)
+g = 1/tan(k*l*lambda_range)/k
+idx = np.where(np.diff(np.sign(f+g)))
+lambdas = np.expand_dims(lambda_range[idx][::2],axis=0)
+
+summation = (sin(a*lambdas)*sin(k*l*lambdas)*sin(k*(l-x)*lambdas)/(lambdas*( a*sin(k*l*lambdas)*sin(k*l*lambdas) + l*sin(a*lambdas)*sin(a*lambdas) ) ) )*exp(-D_PyC*np.power(lambdas,2)*t.transpose() )
+sums = np.sum(summation,axis=1)
+
+analytical_conc = c0*(D_PyC*(l-x)/(l*D_PyC + a*D_SiC)  - 2*sums)*np.ones(sums.shape)
```
Comment (RemDelaporteMathurin) [inline test/tests/ver-1e/comparison_ver-1e.py:40]: I know this is not what's done in the TMAP4 and TMAP7 reports but I would suggest comparing the computed solution to the analytical solution in several x positions, not just one far away from the interface.

Before this fix, the analytical solution and the numerical solution agreed for one position but not for the others.

The comparison plot could look like this:

image

File: test/tests/ver-1e/comparison_ver-1e.py (hunk 1 above)
Comment (simopier) [reply test/tests/ver-1e/comparison_ver-1e.py:21]: Good catch, @RemDelaporteMathurin !

File: test/tests/ver-1e/comparison_ver-1e.py (hunk 8 above)
Comment (simopier) [reply test/tests/ver-1e/comparison_ver-1e.py:40]: That's a good point, @RemDelaporteMathurin. @ChaitanyaBhave-e, could you add a few more spots (in addition to the TMAP4 and TMAP7 spots)? 

I also want to point out that because we have tests with the exodus files generated by TMAP8, the tests already track the position at each point of he mesh and make sure that new changes do not affect that. Coupled with comparing against the analytical solution in a handful of points, we're well covered.

File: test/tests/ver-1e/tests (hunk 6 above)
Comment (ChaitanyaBhave-e) [reply test/tests/ver-1e/tests:11]: map = False makes results not match between parallel and serial runs. I'm not sure sure what match_ids does, but I think it's something similar. I don't know why this is needed, but didn't want to touch what I don't understand.

File: test/tests/ver-1e/comparison_ver-1e.py (hunk 8 above)
Comment (ChaitanyaBhave-e) [reply test/tests/ver-1e/comparison_ver-1e.py:40]: One point here, TMAP4 and 7 do not provide an analytical solution for the IPyC side of the interface. I could try to find the original derivations from Carslaw and Jaeger, which should have solutions for both sides. But for now even if we plot a few more points it's not a proper analytical solution verification - it's equivalent to the exodiff we are already performing. Another option would potentially be to autogenerate a gif from the simulation output and add it to the site documentation.

Comment (RemDelaporteMathurin) [reply test/tests/ver-1e/comparison_ver-1e.py:46]: Oh I didn't know this my bad. It's true that we only compared FESTIM to TMAP8 here not an analytical solution. 

Since there is continuity of concentration at the interface, it shouldn't be too hard to solve it analytically no?

File: doc/content/verification_and_validation/ver-1e.md [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 ## Test Description
 
-This verification problem is taken from [!cite](ambrosek2008verification). In this problem a composite structure of PyC and SiC is modeled with a constant concentration boundary condition of the free surface of PyC and zero concentration boundary condition on the free surface of the SiC. The steady state solution for the PyC is given as:
+This verification problem is taken from [!cite](longhurst1992verification, ambrosek2008verification). In this problem a composite structure of PyC and SiC is modeled with a constant concentration boundary condition of the free surface of PyC and zero concentration boundary condition on the free surface of the SiC. The steady state solution for the PyC is given as:
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:7]:
```suggestion
This verification problem is taken from [!cite](longhurst1992verification, ambrosek2008verification). In this problem, a composite structure of PyC and SiC is modeled with a constant concentration boundary condition of the free surface of PyC and zero concentration boundary condition on the free surface of the SiC. The steady state solution for the PyC is given as:
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 10]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
 
 and $\lambda_n$ are the roots of
 
-$tan(\lambda a) + k \; tan(k l \lambda) = 0$
+$\frac{1}{tan(\lambda a)} + \frac{1}{k \; tan(k l \lambda)} = 0$
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:48]:
```suggestion
\begin{equation}
\label{eqn:roots}
    \frac{1}{tan(\lambda a)} + \frac{1}{k \; tan(k l \lambda)} = 0.
\end{equation}
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 11]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
...
 
+NOTE: The above expression for the roots of $\lambda$ is obtained from [!cite](longhurst1992verification). The equation presented in [!cite](ambrosek2008verification) has a typographical error and gives incorrect results.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:50]:
```suggestion
!alert warning title=Typo in [!cite](ambrosek2008verification)
[eqn:roots] for the roots of $\lambda$ is obtained from [!cite](longhurst1992verification). The equation presented in [!cite](ambrosek2008verification) has a typographical error and gives incorrect results.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 12]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
...
 ## Results
 
-[ver-1e_comparison_dist] shows the comparison of the TMAP8 calculation and the analytical solution for concentration after steady-state is reached.
+[ver-1e_comparison_dist] shows the comparison of the TMAP8 calculation and the analytical solution for concentration after steady-state is reached. The plots show the TMAP8 and analytical solution comparisons for both the TMAP4 case ($l$ = 63 $\mu m$) and the TMAP7 case ($l$ = 66 $\mu m$). 
 
 !media figures/ver-1e_comparison_dist.png
     style=width:50%;margin-bottom:2%
     id=ver-1e_comparison_dist
-    caption=Comparison of TMAP8 calculation with the analytical solution
+    caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the error between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:59]:
```suggestion
    caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the root mean square percentage error (RMSPE) between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
$R$ is often reserved for RMSE, maybe use RMSPE (root mean square percentage error) in the figures.

File: doc/content/verification_and_validation/ver-1e.md [hunk 13]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
...
 
-For transient solution comparison, the concentation at a point, which is 15.75 $\mu m$ away from the PyC-SiC interface into the SiC layer, is obtained using the TMAP code as well as analytically. [ver-1e_comparison_time] shows comparison of the TMAP calculation with the analytical solution for this transient case. There is good agreement between TMAP and the analytical solution for both steady state as well as transient cases.
+For transient solution comparison, the concentation at a point, which is $x$ $\mu m$ away from the PyC-SiC interface into the SiC layer, is obtained using the TMAP code as well as analytically. [ver-1e_comparison_time] shows comparison of the TMAP calculation with the analytical solution for this transient case. In the TMAP4 case, $x$ = 8 $\mu m$, and in the TMAP7 case $x$ = 15.75 $\mu m$. There is good agreement between TMAP and the analytical solution for both steady state as well as transient cases. In both cases, the root mean square error (RMSE) percent error is under 0.2 %.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:61]:
```suggestion
For transient solution comparison, the concentation at a point, which is $x$ $\mu m$ away from the PyC-SiC interface into the SiC layer, is obtained using the TMAP code as well as analytically. [ver-1e_comparison_time] shows comparison of the TMAP calculation with the analytical solution for this transient case. In the TMAP4 case, $x$ = 8 $\mu m$, and in the TMAP7 case $x$ = 15.75 $\mu m$. There is good agreement between TMAP and the analytical solution for both steady state as well as transient cases. In both cases, the root mean square percentage error (RMSPE) is under 0.2 %. 
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 14]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
...
+For transient solution comparison, the concentation at a point, which is $x$ $\mu m$ away from the PyC-SiC interface into the SiC layer, is obtained using the TMAP code as well as analytically. [ver-1e_comparison_time] shows comparison of the TMAP calculation with the analytical solution for this transient case. In the TMAP4 case, $x$ = 8 $\mu m$, and in the TMAP7 case $x$ = 15.75 $\mu m$. There is good agreement between TMAP and the analytical solution for both steady state as well as transient cases. In both cases, the root mean square error (RMSE) percent error is under 0.2 %. 
 
 !media figures/ver-1e_comparison_time.png
     style=width:50%;margin-bottom:2%
     id=ver-1e_comparison_time
-    caption=Comparison of TMAP8 calculation with the analytical solution
+    caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the error between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:66]:
```suggestion
    caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 15]
```diff
@@ -4,52 +4,179 @@
...
+D_SiC = 2.622e-11       # diffusivity in SiC (m^2/s)
+
+x = tmap_distance_tmap4
+PyC_conc = c0*(1 + (x/l)*((a*D_PyC)/(a*D_PyC + l*D_SiC) - 1 ) )
+SiC_conc = c0*(((a+l-x)/l)*(a*D_PyC)/(a*D_PyC + l*D_SiC) )
+analytical_conc_tmap4 = (x<a)*PyC_conc+(x>=a)*SiC_conc
+
+RMSE = np.sqrt(np.mean((tmap_conc_tmap4-analytical_conc_tmap4)**2))
+err_percent = RMSE*100/np.mean(analytical_conc_tmap4)
+ax.text(50, 15, 'R = %.2f '%err_percent+'% \n(TMAP4)',fontweight='bold')
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:157]:
```suggestion
ax.text(50, 15, 'RMSPE = %.2f '%err_percent+'% \n(TMAP4)',fontweight='bold')
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 16]
```diff
@@ -4,52 +4,179 @@
...
+PyC_conc = c0*(1 + (x/l)*((a*D_PyC)/(a*D_PyC + l*D_SiC) - 1 ) )
+SiC_conc = c0*(((a+l-x)/l)*(a*D_PyC)/(a*D_PyC + l*D_SiC) )
+analytical_conc_tmap7 = (x<a)*PyC_conc+(x>=a)*SiC_conc
 
-analytical_sol = pd.read_csv("./analytical_u_vs_x_steadyState.csv")
-analytical_distance = analytical_sol['x']*1e6
-analytical_conc = analytical_sol['u']
+RMSE = np.sqrt(np.mean((tmap_conc_tmap7-analytical_conc_tmap7)**2))
+err_percent = RMSE*100/np.mean(analytical_conc_tmap7)
+ax.text(80, 25, 'R = %.2f '%err_percent+'% \n(TMAP7)',fontweight='bold')
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:173]:
```suggestion
ax.text(80, 25, 'RMSPE = %.2f '%err_percent+'% \n(TMAP7)',fontweight='bold')
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 17]
```diff
@@ -4,52 +4,179 @@
...
+                        # where we compare analytical and numerical model concentration predictions (m)
+summation = (sin(a*lambdas)*sin(k*l*lambdas)*sin(k*(l-x)*lambdas)/(lambdas*( a*sin(k*l*lambdas)*sin(k*l*lambdas) + l*sin(a*lambdas)*sin(a*lambdas) ) ) )*exp(-D_PyC*np.power(lambdas,2)*t.transpose() )
+sums = np.sum(summation,axis=1)
+
+analytical_conc_tmap4 = c0*(D_PyC*(l-x)/(l*D_PyC + a*D_SiC)  - 2*sums)*np.ones(sums.shape)
+
+idx = np.where(tmap_time_tmap4>=t0)[0]
+RMSE = np.sqrt(np.mean((tmap_conc_tmap4[idx]-analytical_conc_tmap4[idx])**2))
+err_percent = RMSE*100/np.mean(analytical_conc_tmap4[idx])
+ax.text(5,40, 'R = %.2f '%err_percent+'% \n(TMAP4)',fontweight='bold')
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:57]:
```suggestion
ax.text(5,40, 'RMSPE = %.2f '%err_percent+'% \n(TMAP4)',fontweight='bold')
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 18]
```diff
@@ -4,52 +4,179 @@
...
+                        # where we compare analytical and numerical model concentration predictions (m)
+summation = (sin(a*lambdas)*sin(k*l*lambdas)*sin(k*(l-x)*lambdas)/(lambdas*( a*sin(k*l*lambdas)*sin(k*l*lambdas) + l*sin(a*lambdas)*sin(a*lambdas) ) ) )*exp(-D_PyC*np.power(lambdas,2)*t.transpose() )
+sums = np.sum(summation,axis=1)
+
+analytical_conc_tmap7 = c0*(D_PyC*(l-x)/(l*D_PyC + a*D_SiC)  - 2*sums)*np.ones(sums.shape)
+
+idx = np.where(tmap_time_tmap7>=t0)[0]
+RMSE = np.sqrt(np.mean((tmap_conc_tmap7[idx]-analytical_conc_tmap7[idx])**2))
+err_percent = RMSE*100/np.mean(analytical_conc_tmap7[idx])
+ax.text(15, 25, 'R = %.2f '%err_percent+'% \n(TMAP7)',fontweight='bold')
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:83]:
```suggestion
ax.text(15, 25, 'RMSPE = %.2f '%err_percent+'% \n(TMAP7)',fontweight='bold')
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 19]
```diff
@@ -4,52 +4,179 @@
...
-analytical_distance = analytical_sol['x']*1e6
-analytical_conc = analytical_sol['u']
+RMSE = np.sqrt(np.mean((tmap_conc_tmap7-analytical_conc_tmap7)**2))
+err_percent = RMSE*100/np.mean(analytical_conc_tmap7)
+ax.text(80, 25, 'R = %.2f '%err_percent+'% \n(TMAP7)',fontweight='bold')
 
-ax.plot(tmap_distance, tmap_conc, label=r"TMAP8", c='tab:gray')
-ax.plot(analytical_distance, analytical_conc,
-        label=r"Analytical", c='k', linestyle='--')
+# ax.plot(tmap_distance_microns, tmap_conc, label=r"TMAP8 (TMAP4 case)", c='tab:gray')
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:175]:
```suggestion
```

File: test/tests/ver-1e/tests [hunk 20]
```diff
@@ -1,12 +1,77 @@
...
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution for the TMAP7 verification case.'
+    exodiff_opts = '-match_ids'
+    map = False
+  []
+  [ver-1e_TMAP4_heavy_csvdiff]
+    type = CSVDiff
+    input = ver-1e.i
+    should_execute = False  # this test relies on the output files from ver-1e_heavy, so it shouldn't be run twice
+    csvdiff = TMAP4.csv
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution to generate CSV data for use in comparisons with the analytic solution over time  for the TMAP4 verification case.'
```
Comment (simopier) [inline test/tests/ver-1e/tests:38]:
```suggestion
    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution to generate CSV data for use in comparisons with the analytic solution over time for the TMAP4 verification case.'
```

File: test/tests/ver-1e/tests [hunk 21]
```diff
@@ -1,12 +1,77 @@
...
+    heavy = true
+    cli_args = 'T_SiC=66e-6 D_ver=15.75e-6 Outputs/exodus/file_base=TMAP7 Outputs/csv/file_base=TMAP7 Outputs/vector_postproc/file_base=TMAP7_vector_postproc'
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution for the TMAP7 verification case.'
+    exodiff_opts = '-match_ids'
+    map = False
+  []
+  [ver-1e_TMAP4_heavy_csvdiff]
+    type = CSVDiff
+    input = ver-1e.i
+    should_execute = False  # this test relies on the output files from ver-1e_heavy, so it shouldn't be run twice
```
Comment (simopier) [inline test/tests/ver-1e/tests:36]:
```suggestion
    should_execute = False  # this test relies on the output files from ver-1e_TMAP4_heavy, so it shouldn't be run twice
```

File: test/tests/ver-1e/tests [hunk 22]
```diff
@@ -1,12 +1,77 @@
...
+    csvdiff = TMAP4.csv
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution to generate CSV data for use in comparisons with the analytic solution over time  for the TMAP4 verification case.'
+    heavy = true
+    prereq = ver-1e_TMAP4_heavy
+  []
+  [ver-1e_TMAP7_heavy_csvdiff]
+    type = CSVDiff
+    input = ver-1e.i
+    should_execute = False  # this test relies on the output files from ver-1e_heavy, so it shouldn't be run twice
```
Comment (simopier) [inline test/tests/ver-1e/tests:45]:
```suggestion
    should_execute = False  # this test relies on the output files from ver-1e_TMAP7_heavy, so it shouldn't be run twice
```

File: test/tests/ver-1e/tests [hunk 23]
```diff
@@ -1,12 +1,77 @@
...
+    csvdiff = TMAP7.csv
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and time step required to match the analytical solution to generate CSV data for use in comparisons with the analytic solution over time for the TMAP7 verification case.'
+    heavy = true
+    prereq = ver-1e_TMAP7_heavy
+  []
+  [ver-1e_TMAP4_heavy_lineplot]
+    type = CSVDiff
+    input = ver-1e.i
+    should_execute = False  # this test relies on the output files from ver-1e_heavy, so it shouldn't be run twice
```
Comment (simopier) [inline test/tests/ver-1e/tests:54]:
```suggestion
    should_execute = False  # this test relies on the output files from ver-1e_TMAP4_heavy, so it shouldn't be run twice
```

File: test/tests/ver-1e/tests [hunk 24]
```diff
@@ -1,12 +1,77 @@
...
+    csvdiff = TMAP4_vector_postproc_line_0548.csv
+    requirement = 'The system shall be able to model transient diffusion through a composite slab with a constant concentration boundary condition as the species source, with the fine mesh and timestep required to match the analytical solution to generate CSV data for use in comparisons with the analytic solution for the profile concentration for the TMAP4 verification case.'
+    heavy = true
+    prereq = ver-1e_TMAP4_heavy
+  []
+  [ver-1e_TMAP7_heavy_lineplot]
+    type = CSVDiff
+    input = ver-1e.i
+    should_execute = False  # this test relies on the output files from ver-1e_heavy, so it shouldn't be run twice
```
Comment (simopier) [inline test/tests/ver-1e/tests:63]:
```suggestion
    should_execute = False  # this test relies on the output files from ver-1e_TMAP7_heavy, so it shouldn't be run twice
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 25]
```diff
@@ -45,23 +45,32 @@ $k$ = $\sqrt{\frac{D_{PyC}}{D_{SiC}}}$
...
+
+The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.2 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation.
+
+!media figures/ver-1e_comparison_time_closeup.png
+    style=width:50%;margin-bottom:2%
+    id=ver-1e_comparison_time_zoomed
+    caption=Zoomed-in view of comparison of TMAP8 calculation with the analytical solution for $t$ < 1 s. The analytical solution shows unphysical predictions close to $t$ = 0 s.
+
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:74]:
```suggestion

## Comments on verification input file versus test file

It is important to note that the input file used to reproduce these results and the input file used as test in TMAP8 are different. Indeed, the input file [/ver-1e.i] has a fine mesh and uses small time steps to accurately match the analytical solutions and reproduce the figures above. To limit the computational costs of the tests, however, the tests run a version of the file with a coarser mesh and larger time steps. More information about the changes can be found in the test specification file for this case [/ver-1e/tests].
```

**Review by simopier — CHANGES_REQUESTED** [review]
Thank you for adding all this documentation and improving the input files and the scripts! Great work.

I have some more comments, but they are mostly minor.

File: test/tests/ver-1e/comparison_ver-1e.py (hunk 8 above)
Comment (simopier) [reply test/tests/ver-1e/comparison_ver-1e.py]: @RemDelaporteMathurin 
We'll table the development of the analytical solution for now. Since this PR focuses on fixing the verification test and improving reproducibility, we'll do that and merge the PR. At that point, the TMAP8 verification case will be more detailed than TMAP4 and TMAP7 (and will combine them). We can revisit the analytical solution later on. 

Moreover, as stated above, the information is saved in the exodus outputs, and deviations from this are tracked.

**Review by simopier — APPROVED** [review]
Looks great to me, @ChaitanyaBhave-e. Thank you for cleaning up this verification case!
